src/opm/output/eclipse/AggregateMSWData.cpp

```diff
@@ -438,6 +438,20 @@ namespace {
             iSeg[baseIndex + Ix::ICDOpenShutFlag] = sicd.ecl_status();
         }
 
+        template <class ISegArray>
+        void assignAICDCharacteristics(const Opm::Segment& segment,
+                                            const std::size_t   baseIndex,
+                                            ISegArray&          iSeg)
+        {
+            using Ix = ::Opm::RestartIO::Helpers::
+                VectorItems::ISeg::index;
+
+            const auto& aicd = segment.autoICD();
+            iSeg[baseIndex + Ix::SegmentType] = -8;
```

Comment:
Magic number alarm!

Reply:
I agree that we do not know why the segment type value is -8, however, we know that the value for an AICD segment is this value. I have added a comment about that to explain.

Reply:
Sure - I understand that it actually is a magic number - in the true sense of the word; my point is just that it should be named as a named constant - e.g.
```
SegmentypeAICD = 8;
```
in the `M` namespace.

Reply:
Good remark: I will use segment.ecl_type_id(), should have used that in the first place!!